src/vs/code/electron-main/app.ts

```diff
@@ -1471,4 +1482,49 @@ export class CodeApplication extends Disposable {
 		// Refs: https://github.com/microsoft/vscode/issues/234064
 		validatedevDeviceId(this.stateService, this.logService);
 	}
+
+	private async moveToApplicationsFolder(accessor: ServicesAccessor): Promise<void> {
+		const dialogMainService = accessor.get(IDialogMainService);
+
+		const { response } = await dialogMainService.showMessageBox({
+			type: 'info',
+			buttons: [
+				localize({ key: 'move', comment: ['&& denotes a mnemonic'] }, "&&Move to Applications folder"),
+				localize({ key: 'doNotMove', comment: ['&& denotes a mnemonic'] }, "&&Cancel")
+			],
+			message: localize('moveToApplicationsFolderWarning', "Current version of {0} is installed outside the Applications folder, would you like to move it ?", this.productService.nameLong),
+			checkboxLabel: localize('remember', "Do not ask again"),
+			cancelId: 1
+		});
+
+		if (response === 0) {
+			try {
+				const result = app.moveToApplicationsFolder({
```

Comment:
Its unclear to me what `moveToApplicationsFolder` does under the hood? Will it trigger our shutdown handlers, for example configure `files.hotExit` to `off`, have a dirty file and trigger the move, would it still confirm shutdown and allow to veto?

Reply:
underneath the equivalent of `app.quit` is called from the runtime to exit the application which should invoke all shutdown related handlers for graceful exit, I haven't tested the above flow, will confirm.